cfgov/unprocessed/js/molecules/Expandable.js

```diff
@@ -0,0 +1,343 @@
+    _setMinHeight();
+    _transitionHeight( _collapseComplete, duration );
+    return this;
+  }
+
+  /**
+   * Transition height property and call a callback function.
+   * Call callback directly if CSS transitions aren't supported.
+   * @param {Function} callback Callback function for completion of transition.
+   * @param {number} duration
+   *   The duration of the sliding animation in milliseconds.
+   */
+  function _transitionHeight( callback, duration ) {
+    if ( _transitionEndEvent ) {
+      _content.addEventListener( _transitionEndEvent, callback );
+      _content.style.transition = 'height ' + duration + 's ease-out';
+    } else {
+      // TODO: Remove callback-return ESLint ignore
+      callback(); // eslint-disable-line callback-return, inline-comments, max-len
+    }
+  }
+
+  /**
+   * Configure the expanded state appearance.
+   */
+  function _setExpandedState() {
+    _dom.classList.add( BASE_CLASS + '__expanded' );
+    _content.setAttribute( 'aria-expanded', 'true' );
+    _target.setAttribute( 'aria-pressed', 'true' );
+    _setStateTo( EXPANDED );
+  }
+
+  /**
+   * Configure the collapsed state appearance.
+   */
+  function _setCollapsedState() {
+    _dom.classList.remove( BASE_CLASS + '__expanded' );
+    _content.setAttribute( 'aria-expanded', 'false' );
+    _target.setAttribute( 'aria-pressed', 'false' );
+    _setStateTo( COLLAPSED );
+  }
+
+  /**
+   * Expand animation has completed.
+   */
+  function _expandComplete() {
+    _content.removeEventListener( _transitionEndEvent, _expandComplete );
+    _setExpandedState();
+    _that.dispatchEvent( 'endExpand', { target: _that } );
+  }
+
+  /**
+   * Collapse animation has completed.
+   */
+  function _collapseComplete() {
+    _content.removeEventListener( _transitionEndEvent, _collapseComplete );
+    _setCollapsedState();
+    _that.dispatchEvent( 'endCollapse', { target: _that } );
+  }
+
+  /**
+   * Handle click of the Expandable target.
+   */
+  function _handleClick() {
+    // Bubble click event outside of the Expandable.
+    _that.dispatchEvent( 'click', { target: _that } );
+    if ( _isCollapsed() || _isExpanded() ) {
+      _that.toggle();
+    }
+  }
+
+  /**
+   * Refresh calculated height of content area.
+   */
+  function _calcHeight() {
+    _contentHeight = _contentAnimated.offsetHeight;
+  }
+
+  /**
+   * Reset the height of the Expandables, when e.g. resizing the window.
+   */
+  function _refreshHeight() {
+    if ( _isExpanded() ) {
+      _setMaxHeight();
+    } else {
+      _setMinHeight();
+    }
+  }
+
+  /**
+   * Calculate and set the height based on the contents' height.
+   */
+  function _setMaxHeight() {
+    _calcHeight();
+    _content.style.height = _contentHeight + 'px';
+  }
+
+  /**
+   * Set the height to zero.
+   */
+  function _setMinHeight() {
+    _content.style.height = '0';
+  }
+
+  /**
+   * @returns {boolean} Whether Expandable is in a collapsed state.
+   */
+  function _isCollapsed() {
```

Comment:
So I was thinking the we use `Object. defineProperty` the same way we are using it [here](https://github.com/cfpb/cfgov-refresh/blob/flapjack/cfgov/unprocessed/js/modules/util/aria-state.js). This way you don't have the setters.